### PR TITLE
[MSFT_Compliance] Standalone build + NuGet packaging

### PR DESCRIPTION
**Pull request by brywang-msft**
OneGet/NuGetProvider should publish a package for OneGet/OneGet to consume.

- Copies of required files from OneGet/OneGet
- Add mt.exe
- Now builds standalone
- Add NuGet packaging helper

### CODE REVIEW

**Comment by jianyunt** [comment]
these changes in .cs files are shown as binary changes so they cannot be reviewed. any reasons?

**Comment by jianyunt** [comment]
> //

Can you clean up the OneGet repo by deleting all these .cs files from OneGet repo?

---
Refers to: Microsoft.PackageManagement/providers/inbox/Common/Extensions/Extensions.cs:1 in 945551c. [](commit_id = 945551c76d127f2fc0a9fdaab662c4f6dad098ab, deletion_comment = False)

File: releasenotes.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# 3.0.0.2
```
Comment (jianyunt) [inline releasenotes.md:1]: >3.0.0.2 [](start = 2, length = 7)

Does 3.0.0.2 work with older version of OneGet? If not, we need to let user know what to do.

File: Sync-OneGetCode.ps1 [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<# Sync dependencies from OneGet repo... #>
+param()
+$paths = @("Microsoft.PackageManagement\providers\inbox\Common\Extensions\Extensions.cs",
+           "Microsoft.PackageManagement\providers\inbox\Common\Utility\*.cs",
+           "Microsoft.PackageManagement\providers\inbox\Common\Version\*.cs",
```
Comment (jianyunt) [inline Sync-OneGetCode.ps1:5]: Delete them all in OneGet

File: Sync-OneGetCode.ps1 [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+           "Microsoft.PackageManagement\Utility\Platform\OSInformation.cs")
```
Comment (jianyunt) [inline Sync-OneGetCode.ps1:6]: >icrosoft.PackageManagement\Utility\Platform\OSInformation.c [](start = 13, length = 59)

we possibly only need a small portion of OSInformation class in NuGet. If we can make it independent from OneGet, we do not need this script.

File: build.ps1 [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
 param(
-    # Example: C:\code\OneGet
-    [Parameter(Mandatory=$true)]
-    [string]
-    $OneGetRepositoryRoot,
-
     [Parameter(Mandatory=$false)]
     [string]
     [ValidateSet('net451', 'netcoreapp2.0', 'netstandard1.6', 'all')]
     $Framework = 'net451',
```
Comment (jianyunt) [inline build.ps1:6]: we should move to netcoreapp 2.1.3 - RTM version of CoreCLR and  Net4.6?

File: pack.ps1 [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<# Package NuGetProvider into a NuGet package #>
+param(
+    [Parameter(Mandatory=$true)]
```
Comment (jianyunt) [inline pack.ps1:3]: when do we call pack.ps1? a part of our build?

**Comment by brywang-msft** [comment]
Closing this as we're going to fix OneGet build and do that package publish first before changing the NuGetProvider build.

**Pull request closed without merging**